`restore-file` - Add support for new PR Files

ndom91 opened the pull request:
<!--

Thanks for contributing! 🍄 Do not ignore this, plz.

1. Does this PR close/fix an existing issue? Write something like: Closes #10

Help us test and visualize this PR:

2. What pages does this PR affect? Include some REAL URLs where you tested the code
3. If applicable, add demonstrative screenshots or gifs

Lastly:

4. Open the PR as draft and review it yourself first. Fix what you find and explain weird code, if necessary.

If you haven't done so yet, check out the Contributing page in the wiki: https://github.com/refined-github/refined-github/wiki/Contributing#metadata-guidelines

-->

Fixes: https://github.com/refined-github/refined-github/issues/8714

Reopening #8715. Good call, the new file name was available in the dropdown on the first party links


- Unfortunately the since the dropdowns are now portaled we can't get the old filename from the file header in the main diff view. That old filename really doesn't seem to be anywhere in the portaled dropdown unfortunately.
- But it looks like the new react UI props are available in a script tag that we can leverage. Was able to grab the old file name from any changes that are labelled "RENAMED" from there. What do you think of using those generally?

For example: 

<details>
<summary><code>Set of react props from PR page script tag</code></summary>

```json
{
    "payload": {
        "isSingleFileMode": false,
        "aliveChannel": "eyJjIjoicHVsbF9yZXF1ZXN0OjI5OTE1ODAzMDUiLCJ0IjoxNzYyNjI1ODAyfQ==--916b0885b1749d5aa89bf6de209b19f96ba2481b586140d4f928ca8b4c565055",
        "bannersData": {
            "banners": {
                "dependabotAutomatedSecurityUpdates": {
                    "render": false
                },
                "pausedDependabotUpdate": {
                    "render": false
                },
                "hiddenCharacterWarning": {
                    "render": false
                }
            }
        },
        "changeGroupGeneration": {
            "canAccessChangeGroups": false,
            "changeGroupFeedbackOptions": [
                {
                    "label": "Groups contain changes that do not belong together",
                    "value": "NOT_RELEVANT"
                },
                {
                    "label": "Groups are harmful or unsafe",
                    "value": "OFFENSIVE_OR_DISCRIMINATORY"
                },
                {
                    "label": "Groups are poorly formatted",
                    "value": "POORLY_FORMATTED"
                },
                {
                    "label": "Groups are not helpful",
                    "value": "NOT_HELPFUL"
                },
                {
                    "label": "Groups miss changes that should be included",
                    "value": "MISSED_CONTENT"
                },
                {
                    "label": "Other",
                    "value": "OTHER"
                }
            ],
            "changeGroups": [],
            "unavailableReason": null,
            "latestChangeGroupOrchestration": null,
            "changeGroupGenerationChannel": "eyJjIjoicHVsbF9yZXF1ZXN0OjI5OTE1ODAzMDU6Y2hhbmdlX2dyb3VwX2dlbmVyYXRpb24iLCJ0IjoxNzYyNjI1ODAyfQ==--cdf22234dc9a046fc1d932d776b71e65d92afae3e6cf57d4dcbe4889d79ad8f2"
        },
        "codeowners": null,
        "commits": [
            {
                "actorLogin": "ndom91",
                "createdAt": "2025-11-08T17:27:03+01:00",
                "messageHeadline": "Dummy commit",
                "oid": "43d8cb6ca432205345a0a881a163f932db7566f5",
                "shortOid": "43d8cb6"
            },
            {
                "actorLogin": "ndom91",
                "createdAt": "2025-11-08T17:30:23+01:00",
                "messageHeadline": "Rename commit",
                "oid": "106f9054bf4d387bca0ac3f2ea07a321726d2d29",
                "shortOid": "106f905"
            },
            {
                "actorLogin": "ndom91",
                "createdAt": "2025-11-08T18:46:06+01:00",
                "messageHeadline": "Discard changes to LICENSE",
                "oid": "ac6c7f8396a502bdb75e6c980ff36d262f88c016",
                "shortOid": "ac6c7f8"
            },
            {
                "actorLogin": "ndom91",
                "createdAt": "2025-11-08T19:10:59+01:00",
                "messageHeadline": "Discard changes to hometime.service",
                "oid": "3888cb117d589c609bab6e245dd65c42cfa0e635",
                "shortOid": "3888cb1"
            }
        ],
        "diffSummaries": [
            {
                "changeType": "MODIFIED",
                "isCodeowner": null,
                "isManifestFile": true,
                "isVendored": false,
                "linesAdded": 1,
                "linesChanged": 2,
                "linesDeleted": 1,
                "markedAsViewed": false,
                "path": "package.json",
                "pathDigest": "7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519",
                "highestAnnotationLevel": null,
                "markersMap": {}
            },
            {
                "changeType": "MODIFIED",
                "isCodeowner": null,
                "isManifestFile": false,
                "isVendored": false,
                "linesAdded": 1,
                "linesChanged": 1,
                "linesDeleted": 0,
                "markedAsViewed": false,
                "path": "tsconfig.json",
                "pathDigest": "b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0",
                "highestAnnotationLevel": null,
                "markersMap": {}
            }
        ],
        "diffContents": [
            {
                "isBinary": false,
                "isSubmodule": false,
                "isTooBig": false,
                "diffLines": [
                    {
                        "type": "HUNK",
                        "blobLineNumber": 0,
                        "position": 0,
                        "displayNoNewLineWarning": false,
                        "text": "@@ -1,6 +1,6 @@",
                        "html": "@@ -1,6 +1,6 @@",
                        "left": 0,
                        "right": 0,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 1,
                        "position": 1,
                        "displayNoNewLineWarning": false,
                        "text": " {",
                        "html": " {",
                        "left": 1,
                        "right": 1,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 2,
                        "position": 2,
                        "displayNoNewLineWarning": false,
                        "text": "   \"name\": \"hometime-server\",",
                        "html": "   <span class=\"pl-ent\">&quot;name&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>hometime-server<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 2,
                        "right": 2,
                        "problems": []
                    },
                    {
                        "type": "DELETION",
                        "blobLineNumber": 3,
                        "position": 3,
                        "displayNoNewLineWarning": false,
                        "text": "-  \"version\": \"1.0.0\",",
                        "html": "-  <span class=\"pl-ent\">\"version\"</span>: <span class=\"pl-s\"><span class=\"pl-pds\">\"</span><span class=\"x x-first x-last\">1</span>.0.0<span class=\"pl-pds\">\"</span></span>,",
                        "left": 3,
                        "right": 2,
                        "problems": []
                    },
                    {
                        "type": "ADDITION",
                        "blobLineNumber": 3,
                        "position": 4,
                        "displayNoNewLineWarning": false,
                        "text": "+  \"version\": \"2.0.0\",",
                        "html": "+  <span class=\"pl-ent\">\"version\"</span>: <span class=\"pl-s\"><span class=\"pl-pds\">\"</span><span class=\"x x-first x-last\">2</span>.0.0<span class=\"pl-pds\">\"</span></span>,",
                        "left": 3,
                        "right": 3,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 4,
                        "position": 5,
                        "displayNoNewLineWarning": false,
                        "text": "   \"description\": \"LED progress bar visualization for work hours and calendar events\",",
                        "html": "   <span class=\"pl-ent\">&quot;description&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>LED progress bar visualization for work hours and calendar events<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 4,
                        "right": 4,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 5,
                        "position": 6,
                        "displayNoNewLineWarning": false,
                        "text": "   \"main\": \"dist/index.js\",",
                        "html": "   <span class=\"pl-ent\">&quot;main&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>dist/index.js<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 5,
                        "right": 5,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 6,
                        "position": 7,
                        "displayNoNewLineWarning": false,
                        "text": "   \"scripts\": {",
                        "html": "   <span class=\"pl-ent\">&quot;scripts&quot;</span>: {",
                        "left": 6,
                        "right": 6,
                        "problems": []
                    }
                ],
                "linesAdded": 1,
                "linesChanged": 2,
                "linesDeleted": 1,
                "oldCommitOid": "611aaf78d8f24679931109aaae3e28c9f02735d0",
                "newCommitOid": "3888cb117d589c609bab6e245dd65c42cfa0e635",
                "oldTreeEntry": {
                    "mode": 100644,
                    "path": "package.json",
                    "lineCount": 27
                },
                "newTreeEntry": {
                    "mode": 100644,
                    "path": "package.json",
                    "lineCount": 27,
                    "isGenerated": false
                },
                "path": "package.json",
                "pathDigest": "7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519",
                "richDiff": {
                    "canToggleRichDiff": true,
                    "defaultToRichDiff": false,
                    "proseDiffHtml": null,
                    "renderInfo": null,
                    "dependencyDiffPath": "/ndom91/calendar-led-strip/dependency-review/611aaf78d8f24679931109aaae3e28c9f02735d0/3888cb117d589c609bab6e245dd65c42cfa0e635/rich_diff?manifest_path=package.json"
                },
                "submodule": null,
                "status": "MODIFIED",
                "truncatedReason": null,
                "diffSize": "",
                "reviewed": false
            },
            {
                "isBinary": false,
                "isSubmodule": false,
                "isTooBig": false,
                "diffLines": [
                    {
                        "type": "HUNK",
                        "blobLineNumber": 0,
                        "position": 0,
                        "displayNoNewLineWarning": false,
                        "text": "@@ -1,6 +1,7 @@",
                        "html": "@@ -1,6 +1,7 @@",
                        "left": 0,
                        "right": 0,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 1,
                        "position": 1,
                        "displayNoNewLineWarning": false,
                        "text": " {",
                        "html": " {",
                        "left": 1,
                        "right": 1,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 2,
                        "position": 2,
                        "displayNoNewLineWarning": false,
                        "text": "   \"compilerOptions\": {",
                        "html": "   <span class=\"pl-ent\">&quot;compilerOptions&quot;</span>: {",
                        "left": 2,
                        "right": 2,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 3,
                        "position": 3,
                        "displayNoNewLineWarning": false,
                        "text": "     \"target\": \"ES2020\",",
                        "html": "     <span class=\"pl-ent\">&quot;target&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>ES2020<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 3,
                        "right": 3,
                        "problems": []
                    },
                    {
                        "type": "ADDITION",
                        "blobLineNumber": 4,
                        "position": 4,
                        "displayNoNewLineWarning": false,
                        "text": "+    \"test\": \"12\",",
                        "html": "+    <span class=\"pl-ent\">&quot;test&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>12<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 3,
                        "right": 4,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 5,
                        "position": 5,
                        "displayNoNewLineWarning": false,
                        "text": "     \"module\": \"commonjs\",",
                        "html": "     <span class=\"pl-ent\">&quot;module&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>commonjs<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 4,
                        "right": 5,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 6,
                        "position": 6,
                        "displayNoNewLineWarning": false,
                        "text": "     \"lib\": [\"ES2020\"],",
                        "html": "     <span class=\"pl-ent\">&quot;lib&quot;</span>: [<span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>ES2020<span class=\"pl-pds\">&quot;</span></span>],",
                        "left": 5,
                        "right": 6,
                        "problems": []
                    },
                    {
                        "type": "CONTEXT",
                        "blobLineNumber": 7,
                        "position": 7,
                        "displayNoNewLineWarning": false,
                        "text": "     \"outDir\": \"./dist\",",
                        "html": "     <span class=\"pl-ent\">&quot;outDir&quot;</span>: <span class=\"pl-s\"><span class=\"pl-pds\">&quot;</span>./dist<span class=\"pl-pds\">&quot;</span></span>,",
                        "left": 6,
                        "right": 7,
                        "problems": []
                    }
                ],
                "linesAdded": 1,
                "linesChanged": 1,
                "linesDeleted": 0,
                "oldCommitOid": "611aaf78d8f24679931109aaae3e28c9f02735d0",
                "newCommitOid": "3888cb117d589c609bab6e245dd65c42cfa0e635",
                "oldTreeEntry": {
                    "mode": 100644,
                    "path": "tsconfig.json",
                    "lineCount": 19
                },
                "newTreeEntry": {
                    "mode": 100644,
                    "path": "tsconfig.json",
                    "lineCount": 20,
                    "isGenerated": false
                },
                "path": "tsconfig.json",
                "pathDigest": "b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0",
                "richDiff": null,
                "submodule": null,
                "status": "MODIFIED",
                "truncatedReason": null,
                "diffSize": "",
                "reviewed": false
            }
        ],
        "fileFilter": {
            "initialState": {
                "fileExtensions": {
                    ".json": 2
                },
                "filterText": "",
                "showDeletedFiles": true,
                "showOnlyManifestFiles": false,
                "showOnlyOwnedFiles": false,
                "showVendoredFiles": true,
                "showViewedFiles": true,
                "unselectedFileExtensions": []
            },
            "menuOptions": {
                "canSeeCodeownersFilter": false,
                "canSeeDeletedFilesFilter": false,
                "canSeeOnlyManifestFilesFilter": true,
                "canSeeVendorFilesFilter": false
            }
        },
        "pullRequest": {
            "author": {
                "login": "ndom91"
            },
            "aliveChannel": "eyJjIjoicHVsbF9yZXF1ZXN0OjI5OTE1ODAzMDUiLCJ0IjoxNzYyNjI1ODAyfQ==--916b0885b1749d5aa89bf6de209b19f96ba2481b586140d4f928ca8b4c565055",
            "baseBranch": "main",
            "commitsCount": 4,
            "comparison": {
                "baseOid": "611aaf78d8f24679931109aaae3e28c9f02735d0",
                "headOid": "3888cb117d589c609bab6e245dd65c42cfa0e635"
            },
            "globalRelayId": "PR_kwDOPhSFV86yT-SR",
            "headBranch": "dummy-branch",
            "headRepositoryName": "calendar-led-strip",
            "headRepositoryOwnerLogin": "ndom91",
            "id": 2991580305,
            "isInAdvisoryRepo": false,
            "mergedBy": null,
            "mergedTime": null,
            "number": 1,
            "pathName": "/ndom91/calendar-led-strip/pull/1",
            "state": "DRAFT",
            "title": "Dummy commit",
            "titleHtml": "Dummy commit",
            "viewerAllowedNonCommentReviewTypes": [],
            "viewerHasViolatedPushPolicy": false,
            "viewerIsCopilotAttributed": false
        },
        "repository": {
            "codespacesEnabled": true,
            "editorEnabled": false,
            "defaultBranch": "main",
            "id": 1041532247,
            "isEnterprise": false,
            "name": "calendar-led-strip",
            "ownerLogin": "ndom91",
            "viewerPermission": "admin"
        },
        "threads": {},
        "urls": {
            "images": {
                "mona-hifive": "/images/mona-hifive.gif"
            },
            "checks": "/ndom91/calendar-led-strip/pull/1/checks",
            "commits": "/ndom91/calendar-led-strip/pull/1/commits",
            "conversation": "/ndom91/calendar-led-strip/pull/1",
            "files": "/ndom91/calendar-led-strip/pull/1/files",
            "walkthrough": "/ndom91/calendar-led-strip/pull/1/walkthrough"
        },
        "user": {
            "canChangeBase": true,
            "canComment": true,
            "currentUserAvatarUrl": "https://avatars.githubusercontent.com/u/7415984?v=4",
            "currentUserLogin": "ndom91",
            "canEditTitle": true,
            "hasCopilotAccess": true,
            "canApplySuggestion": true,
            "isFileTreeExpanded": true,
            "lastReviewOid": null,
            "shouldShowViewedFilesCount": true,
            "tabSize": 2,
            "viewedFilesCount": 0,
            "viewSettings": {
                "hideWhitespace": false,
                "lineSpacing": "compact",
                "showChangeGroups": true,
                "splitPreference": "split",
                "commentsPreference": "visible"
            }
        },
        "userNotices": [
            {
                "name": "compact_diff_lines",
                "dismissed": true
            },
            {
                "name": "single_file_mode",
                "dismissed": true
            }
        ],
        "viewerPendingReview": {
            "id": null,
            "comments": []
        },
        "markers": {
            "threads": {},
            "annotations": {}
        },
        "pageLimits": {
            "annotationsLimit": 50,
            "annotationsLimitExceeded": false,
            "filesLimit": 1000,
            "filesLimitExceeded": false,
            "reviewCommentsPerThreadLimit": 20,
            "reviewThreadsLimit": 40,
            "reviewThreadsLimitExceeded": false
        }
    },
    "title": "Dummy commit by ndom91 · Pull Request #1 · ndom91/calendar-led-strip",
    "appPayload": {
        "helpUrl": "https://docs.github.com",
        "refListCacheKey": "v0:1762619230.0",
        "enabled_features": {
            "accessible_code_button": true,
            "copilot_workspace": false,
            "cotd_derive_thread_previews": false,
            "batch_suggested_changes": true,
            "prx_dynamic_split_pref": false,
            "prx_files": true,
            "prx_files_lite_throttle": false,
            "prx_files_medium_throttle": false,
            "pulls-files-skip-prefer-any-solo-entry": false,
            "generate_groups_on_files_view": false,
            "react_data_router_pull_request_files": false,
            "cotd_ui": true,
            "automated_review_comment_simplification": true
        }
    }
}
```


</details>

- Tested both discard normal changes as well as renames and they both work 🙌 

## Test URLs

- https://github.com/refined-github/sandbox/pull/16/files

## Screenshot

Still rendering in old view and renders nicely in new view now as well:

<table>
<tr>
<td align="center"> 

<strong>Old View</strong>

</td>
<td align="center">

<strong>New View</strong>

</td>
</tr>
<tr>
<td>

<img width="380" height="451" alt="CleanShot 2025-10-21 at 18 33 35" src="https://github.com/user-attachments/assets/dfb8d376-9279-42d1-aa75-5142db5ed3c0" />

</td>
<td>

<img width="424" height="345" alt="CleanShot 2025-10-21 at 18 32 46" src="https://github.com/user-attachments/assets/b9264819-c83d-43fc-b84c-6ada61a453a5" />


</td>
</tr>
</table>

fregante commented on source/features/restore-file.tsx:
```diff
@@ -87,15 +88,48 @@ async function discardChanges(progress: (message: string) => void, originalFileN
+			.parentNode
+			?.parentNode
+			?.querySelector('li[data-variant="danger"] a')!
+			.href;
```
1. Don't use `?.` but `!.` or else errors will just be silenced. We want errors to happen as early as possible
3. Use `parentElement` instead of `parentNode`
3. Use `.closest()` when traversing the DOM more than one `parent` away

ndom91 replied:
This is strange, but I can't seem to get `closest()` to find the element when selecting it via `'li[data-variant="danger"]'` (i.e. `$0.closest('li[data-variant="danger"]')`).

`$0` being the `menuItem` (i.e. the anchor tag in the `li` of our new btn). `closest` is supposed to walk up the tree as well, right? I also can't seem to get it to work adding another `.parentElement` before it, so that it only has to go up one level.

Alternatively, would you prefer this?

```js
const fileUrl = menuItem
	.parentElement!
	.parentElement!
	.querySelector('li[data-variant="danger"] a')!
	.href;
```


<img width="653" height="467" alt="CleanShot 2025-11-09 at 13 20 33" src="https://github.com/user-attachments/assets/d9c0577d-1b31-409f-8716-3542f82e4439" />

fregante replied:
If you can't get it to work, then the selector is wrong. `$0.closest(selector)` must work. You can't use `.rgh-restore-file`, how does that class get there in the first place?

ndom91 commented:
Okay, I've done the following:
- Extracted `getFilenames` into its own helper for the pr files page
- Fixed the removal of the discarded file element. This was also a bit trickier in the new portaled setup :/
- Used octicon instead of hardcoded svg
- Replaced `?.` with `!.` to shush TS but still error

Let me know what you think 🙏

fregante commented on source/github-helpers/pr-branches.ts:
```diff
@@ -76,3 +77,42 @@ export function getBranches(): {base: PrReference; head: PrReference} {
 	};
 }
+
+export function getFilenames(menuItem: HTMLElement): {original: string; new: string} {
```
This has nothing to do with "PR branches", it should live in `restore-file.tsx`

fregante commented on source/github-helpers/pr-branches.ts:
```diff
@@ -76,3 +77,42 @@ export function getBranches(): {base: PrReference; head: PrReference} {
+		const fileUrl = menuItem
+			.parentElement!
+			.parentElement!
+			.querySelector('li[data-variant="danger"] a')!
```
Note that .closest() only replaces .parentElement.parentElement, not querySelector. So you're most likely looking for `.closest('ul').querySelector('li[data-variant="danger"] a')`

Also I think that you can't delete a deleted file, so this element won't exist on deleted files, so `restore-file` cannot restore deleted files. You might need to get the URL from `View file` instead. You can `querySelector` its `svg` to make it easier

ndom91 replied:
Nice thanks, `li:has(svg.octicon-eye)` did the trick

fregante commented on source/features/restore-file.tsx:
```diff
@@ -86,43 +88,72 @@ async function discardChanges(progress: (message: string) => void, originalFileN
-			Discard changes
-		</button>,
-	);
+	const content = editFile.tagName === 'LI'
```
Instead of assigning it, just attach it. Also place the old one in a condition so it can be removed cleanly next year 

```js
// TODO: Drop support for old view in June 2025
if (editFile.tagName === 'A') {
	editFile.after(
		<button
			className="pl-5 dropdown-item btn-link rgh-restore-file"
			role="menuitem"
			type="button"
		>
			Discard changes
		</button>
	);

  return;
}

etc
```

fregante replied:
> Also place the old one in a condition so it can be removed cleanly next year

This was not addressed and it applies to both conditions. It should ideally look like this:


```js
if (condition) {
  // old code
  return
}

// new code
```

so next year we apply this diff:

```diff
- if (condition) {
-   // old code
-   return
- }

  // new code
```

and you keep the git-blame for that `// new code` line.

Also keep this in mind: https://eslint.org/docs/latest/rules/no-else-return (our lint config is broken right now)

ndom91 commented on source/features/restore-file.tsx:
```diff
@@ -86,43 +88,72 @@ async function discardChanges(progress: (message: string) => void, originalFileN
+	const filesWrapper = $('div[class^="prc-PageLayout-Content-"] div[data-hpc="true"]');
+	if (filesWrapper) {
+		const fileElement = [...filesWrapper.children].find(child =>
+			child.textContent?.includes(filenames.original),
```
Hmm where did you find that section of code in your screenshot? I can't seem to find anything similar on my dummy PR (https://github.com/ndom91/calendar-led-strip/pull/1/files)

ndom91 commented:
Okay, got all the comments squared away (hopefully without wiping any other previous changes of yours haha) except for the exact match on the title thing. Can you tell me where you found that snippet in your screenshot? Wasn't able to find it on my side 🤔

ndom91 commented on source/features/restore-file.tsx:
```diff
@@ -87,43 +132,86 @@ async function discardChanges(progress: (message: string) => void, originalFileN
+		// Remove file element in list as well as portaled dropdown menu
+		fileElement!.remove();
+		menuItem.closest('div[data-focus-trap="active"]')!.remove();
+	} else {
```
Yeah hmm, its mostly due to grabbing too much text via `textContent` on the file wrapper element. Can probably be cut down a bit by targetting that file header element more specifically and grabbing `textContent` on only it.

ndom91 replied:
Okay, was able to simplify it a bit. Unfortunately there isn't a `:has-text()` selector 😂

ndom91 commented:
@fregante friendly reminder 😊

fregante commented on source/features/restore-file.tsx:
```diff
@@ -1,15 +1,61 @@
+	// Get the new filename from the "View File" button href
+	const urlRegex = new RegExp(
+		`https://github.com/${head?.nameWithOwner.replaceAll('/', '/')}/blob/[^/]+/`,
+	);
```
We have GitHubFileURL for parsing and creating URLs

fregante commented on source/features/restore-file.tsx:
```diff
@@ -86,47 +132,95 @@ async function discardChanges(progress: (message: string) => void, originalFileN
+		}
+
+		return filename === filenames.original;
+	}) as HTMLElement;
```
We already have a `handleMenuOpening` listener that can be used to track the last clicked file so we don't have to look for it again. Try saving it to a global on click, e.g. 

```js
let focusedFile: HTMLElement;

function handleMenuOpening(event): void {
  focusedFile = event.target.find the container that we want to .remove();
  
  etc
}
```

ndom91 commented:
> > I think two of those most recent comments are on old versions of the PR and had been addressed
> 
> No, you can see my PR comes exactly after my PR merge commit here: [refined-github/refined-github/pull/8749/files/ac12327d37afd8197e7257b755f46fe93ea93b4d](https://github.com/refined-github/refined-github/pull/8749/files/ac12327d37afd8197e7257b755f46fe93ea93b4d), where you can see this piece of code:
> 
> ```js
> else {
> 	// TODO: Drop support for old view in June 2026
> 	const [originalFileName, newFileName = originalFileName] = menuItem
> 		.closest('[data-path]')!
> 		.querySelector('.Link--primary')!
> 		.textContent
> 		.split(' → ');
> 
> 	return {original: originalFileName, new: newFileName};
> }
> ```
> 
> if you say things like this, it makes me not want to review your PRs.
> 
> > It won't let me reply to them anymore
> 
> You can, refer to #5394 for an explanation of GitHub's UI in this case.

Dude you saw my screenshot, right? It said "Outdated" (in yellow, indicating that the code doesnt exist anymore in HEAD as far as I know) not "Duplicate". I said "it wont let me reply" simply as a statement of fact to explain why I'm writing a new reply instead of replying inline to your comments.

fregante commented:
I don't know what to tell you, if I request a change and the change isn't made, what am I supposed to do? Disregard the UI and look at the code: does the code follow the suggestion I already asked you twice to apply?

What would you do if you have to ask 3 times the same thing and then be told to work on your communication?

Thanks for your work, but I'm closing this for now
